admin/services/headteacher-declaration.service.js

```diff
@@ -41,18 +43,22 @@ headteacherDeclarationService.findPupilBySlugAndDfeNumber = async (urlSlug, dfeN
 /**
  * Fetch pupils and return eligibility to generate HDF
  * @param dfeNumber
+ * @param checkEndDate
+ * @param timezone
  * @returns {Array}
  */
-headteacherDeclarationService.getEligibilityForSchool = async (dfeNumber) => {
-  const pupils = await pupilDataService.sqlFindPupilsWithStatusByDfeNumber(dfeNumber)
-  // check the attendance codes for pupils that don't have the completed status
-  const ids = pupils.filter(p => p.code !== pupilStatusService.STATUS_CODES.COMPLETED).map(p => p.id)
-  if (ids.length === 0) {
-    return true
-  }
-  const pupilAttendance = await pupilAttendanceDataService.findByPupilIds(ids)
-  // check if all pupils that don't have the completed status, have an attendance reason
-  return pupilAttendance.length === ids.length
+headteacherDeclarationService.getEligibilityForSchool = async (dfeNumber, checkEndDate, timezone) => {
+  if (!checkEndDate) {
+    throw new Error('Check end date missing or not found')
+  }
+  const currentDate = moment.tz(timezone || config.DEFAULT_TIMEZONE)
+  if (currentDate.isBefore(checkEndDate)) {
+    const ineligiblePupilsCount = await headteacherDeclarationDataService.sqlFindPupilsBlockingHdfBeforeCheckEndDate(dfeNumber)
+    return ineligiblePupilsCount === 0
+  } else {
+    const ineligiblePupilsCount = await headteacherDeclarationDataService.sqlFindPupilsBlockingHdfAfterCheckEndDate(dfeNumber)
+    return ineligiblePupilsCount === 0
```

Comment:
Duplicate code: `return ineligiblePupilsCount === 0` - perhaps refactor to remove the duplicate.

Reply:
@jon-shipley restructured as suggested.